incubator/binding-kafka-grpc/src/main/java/io/aklivity/zilla/runtime/binding/kafka/grpc/internal/stream/KafkaGrpcRemoteServerFactory.java

```diff
@@ -0,0 +1,1994 @@
+    private final KafkaBeginExFW kafkaBeginExRO = new KafkaBeginExFW();
+    private final KafkaDataExFW kafkaDataExRO = new KafkaDataExFW();
+    private final GrpcQueueMessageFW queueMessageRO = new GrpcQueueMessageFW();
+
+    private final GrpcBeginExFW.Builder grpcBeginExRW = new GrpcBeginExFW.Builder();
+    private final GrpcDataExFW.Builder grpcDataExRW = new GrpcDataExFW.Builder();
+    private final GrpcResetExFW.Builder grpcResetExRW = new GrpcResetExFW.Builder();
+    private final GrpcAbortExFW.Builder grpcAbortExRW = new GrpcAbortExFW.Builder();
+
+    private final KafkaBeginExFW.Builder kafkaBeginExRW = new KafkaBeginExFW.Builder();
+    private final KafkaDataExFW.Builder kafkaDataExRW = new KafkaDataExFW.Builder();
+    private final GrpcQueueMessageFW.Builder queueMessageRW = new GrpcQueueMessageFW.Builder();
+
+    private final Long2ObjectHashMap<KafkaGrpcBindingConfig> bindings;
+    private final BufferPool bufferPool;
+    private final MutableDirectBuffer writeBuffer;
+    private final MutableDirectBuffer extBuffer;
+    private final BindingHandler streamFactory;
+    private final LongUnaryOperator supplyInitialId;
+    private final LongUnaryOperator supplyReplyId;
+    private final LongSupplier supplyTraceId;
+    private final Signaler signaler;
+    private final int grpcTypeId;
+    private final int kafkaTypeId;
+    private long reconnectAt = NO_CANCEL_ID;
+    private long lastBindingId;
+
+
+    public KafkaGrpcRemoteServerFactory(
+        KafkaGrpcConfiguration config,
+        EngineContext context)
+    {
+        this.bufferPool = context.bufferPool();
+        this.writeBuffer = context.writeBuffer();
+        this.extBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
+        this.streamFactory = context.streamFactory();
+        this.supplyInitialId = context::supplyInitialId;
+        this.supplyReplyId = context::supplyReplyId;
+        this.signaler = context.signaler();
+        this.supplyTraceId = context::supplyTraceId;
+        this.bindings = new Long2ObjectHashMap<>();
+        this.grpcTypeId = context.supplyTypeId(GRPC_TYPE_NAME);
+        this.kafkaTypeId = context.supplyTypeId(KAFKA_TYPE_NAME);
+    }
+
+    @Override
+    public void attach(
+        BindingConfig binding)
+    {
+        KafkaGrpcBindingConfig newBinding = new KafkaGrpcBindingConfig(binding);
+        lastBindingId = binding.id;
+        bindings.put(lastBindingId, newBinding);
+
+        this.reconnectAt = signaler.signalAt(
+            currentTimeMillis(),
+            SIGNAL_INITIATE_KAFKA_STREAM,
+            this::onKafkaStreamInitializationSignal);
+    }
+
+    @Override
+    public void detach(
+        long bindingId)
+    {
+        bindings.remove(bindingId);
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer grpc)
+    {
+        return NOOP;
```

Comment:
This should be `null` not `NOOP`, right?

I think we want to `RESET` any attempt to send `BEGIN (initial)` into `REMOTE_SERVER` handler.

Reply:
Yeah